### ISSUE TITLE
Crash on update of legend text

### ISSUE THREAD

**Issue by BeeSee20Eye35** [highfestiva/finplot#254]
Hi

Still loving Finplot, thanks for the continued upkeep. 

I have a window with two Finplot plots - see screenshot : top one has multiple data sets including candlesticks and line data. 
The lower plot has just one set of line data. 

![image](https://user-images.githubusercontent.com/68307682/132758990-f9c44145-6b9f-4183-a435-2a9218b7ab54.png)

The lower plot is one of several indicators I want to switch between. I want the legend to update with the relevant indicator name.

However, I cannot find a way to update the plot legend. 
I have tried : 
1. Using obtaining the label returned by creating the legend in this way : 
legend_label = fplt.add_legend('Info', ax=ax2)

And using setText :
legend_label.setText(indicator_name)

This causes the application to crash.

I have also tried setting the name element of the opts for the plot or the label to the new name. 
indicator_plot.opts['name'] = indicator_name
or
legend_label.opts['name'] = indicator_name

Neither of these causes a crash but they also do not result in an updated legend being displayed. 

Please could you tell me the correct way to update a legend text that already exists?

Thanks

**Comment by highfestiva** [comment]
See example snp500 for how to update legend.

**Comment by highfestiva** [comment]
@BeeSee20Eye35 ping

**Comment by BeeSee20Eye35** [comment]
Thank you. 
I did try that. That was the way I set it up in the first place. Unfortunately, that was what was causing the fatal error. 
It is very hard to reproduce the issue in an example I could share without a lot of additional work so I suspect I will not be able to give more information at this time to facilitate problem solving. 
For the foreseeable future I have removed the legend to avoid having either fatal crashes or a legend that does not update with the data. 

Thank you once again.
